melodies_monet/driver.py

```diff
@@ -619,14 +732,106 @@ def open_model_files(self, time_interval=None, control_dict=None):
+                
+                varmap = self.extra_calc["dewpoint"]
+                self.obj=dewpoint(self.obj, varmap = varmap)
+                #print(self.obj)
+
+            if "rel_hum" in self.extra_calc:
+                print("Calculating modeled relative humidity...")
+                from .util.metcalc import relh
+
+                varmap = self.extra_calc["rel_hum"]
+                self.obj=relh(self.obj, varmap = varmap)
+                #print(self.obj)
+
+            if "windspeed" in self.extra_calc:
+                print("Calculating modeled windpseed...")
+                from .util.metcalc import wspd
+
+                varmap = self.extra_calc["windspeed"]
+                self.obj=wspd(self.obj, varmap = varmap)
+                #print(self.obj)
+                
+            if "winddir" in self.extra_calc:
+                print("Calculating modeled wind direction...")
+                from .util.metcalc import wdir
+
+                varmap = self.extra_calc["winddir"]
+                self.obj=wdir(self.obj, varmap = varmap)
+                #print(self.obj)   
+
+            if "wind_barb" in self.extra_calc:
+                print("Calculating model wind barbs...")
+                u_comp = self.extra_calc.get('wind_barb', {}).get("u_comp", None)
+                v_comp = self.extra_calc.get('wind_barb', {}).get("v_comp", None)         
+            
+            if "ptemp_mod" in self.extra_calc:
+                print("Calculating modeled potential temperature...")
+                from .util.metcalc import ptemp
+            
+                varmap = self.extra_calc["ptemp_mod"]
+
+                # Extract a single vertical column from the model field 
+                # for boundary layer calc
+                pres = self.obj[varmap.get("pres", "pressure_model")]
+                temp = self.obj[varmap.get("temp", "temperature_k")]
+
+                center_y = pres.shape[2] // 2
+                center_x = pres.shape[3] // 2
+                time_idx = 0  
+                
+                pres_column = pres[time_idx, :, center_y, center_x]
+                temp_column = temp[time_idx, :, center_y, center_x]
+
+                column_obj = {
+                    "pressure_model": pres_column,
+                    "temperature_k": temp_column
+                }
+                
+                # Create a temporary object with just this column
+                result = ptemp(
+                    column_obj,
+                    varmap={"pres": "pressure_model", 
+                            "temp": "temperature_k"},
+                    output_key="ptemp_mod"
+                )
+            
+                #Run ptemp on the extracted column
+                self.obj = ptemp(
+                    self.obj,
+                    varmap={"pres": "pressure_model", 
+                            "temp": "temperature_k"},
+                    output_key="ptemp_mod"
+                )
+    
```

Comment:
Don't you want to use varmap here from the yaml file options like you do with the others. This seems a bit different than what you did with the observations. Maybe making these consistent would be useful.